contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -118,6 +141,50 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     emit WatchlistUpdated();
   }
 
+  /// @notice Adds a new address to the watchlist
+  /// @param targetAddress the address to be added to the watchlist
+  /// @param dstChainSelector carries a non-zero value in case the targetAddress is an onRamp, otherwise it carries a 0
+  /// @dev this function has to be compatible with the event onRampSet(address, dstChainSelector) emitted by
+  /// the CCIP router. Important detail to know is this event is also emitted when an onRamp is decomissioned,
+  /// in which case it will carry the proper dstChainSelector along with the 0x0 address
+  function addToWatchList(address targetAddress, uint64 dstChainSelector) public onlyRoleOrAdminRole(EXECUTOR_ROLE) {
+    if (s_targets[targetAddress].isActive) revert DuplicateAddress(targetAddress);
+    address oldAddress = s_onRampAddresses[dstChainSelector];
+    // if targetAddress is an existing onRamp, there's a need of cleaning the previous onRamp associated to this dstChainSelector
+    // there's no need to remove any other address that's not an onRamp
+    if (dstChainSelector > 0 && bytes(abi.encodePacked(oldAddress)).length > 0) {
```

Comment:
Not exactly I think, on this line I want to make sure the dstChainSelector is positive and the oldAddress contains bytes (it exists in the mapping s_onRampAddresses)
If the address does not exist on s_onRampAddresses I assume oldAddress would be 0 bytes, not address(0).

Reply:
unset values in solidity always default to their "zero value" - for for address types that is `address(0)`. So even if the underlying storage has never been set for a particular address storage variable, it will _still_ be equal to `address(0)`.

Reply:
Gotcha! Thanks for educating. :)

Reply:
Oh I think we should still make this change! I think `oldAddress != address(0)` is cleaner